database.rules.json

```diff
@@ -3,24 +3,57 @@
     "users": {
       "$uid": {
         ".read" : "auth !== null && auth.uid === $uid",
-        ".write" : "auth !== null && auth.uid === $uid"
+        "allCalendarIds" : {
+          //".write" : "true",
+          "$calendarId" : {
+            ".write" : "auth !== null && auth.uid === $uid && (!data.exists() || root.child('calendars/' + $calendarId + '/writers/' + $uid).exists())",
+            ".validate" : "newData.isString() && newData.val().length > 0"
+          }
+        },
+        "currentActiveCalendar" : {
+            ".write" : "auth !== null && auth.uid === $uid && (!data.exists() || root.child('calendars/' + newData.val() + '/writers/' + auth.uid).exists())",  
+            ".validate" : "newData.isString() && newData.val().length > 0"
+        }
       }
     },
     "calendars":{
       "$calendarId": {
         "calendarState": {
-          ".read" : "auth !== null && data.parent().child('readers/' + auth.uid).exists()",
-          ".write" : "auth !== null && data.parent().child('writers/' + auth.uid).exists()"
-        },
+          ".read" : "auth !== null && (!data.exists() || root.child('calendars/' + $calendarId + '/readers/' + auth.uid).exists())",
+          ".write" : "auth !== null && (!data.exists() || root.child('calendars/' +$calendarId + '/writers/' + auth.uid).exists())",
+          "startDateString": {
+            ".validate" : "newData.isString() && newData.val().length > 0 && newData.val().matches(/\\d{8}/)"
+          },
+          "endDateString": {
+            ".validate" : "newData.isString() && newData.val().length > 0 && newData.val().matches(/\\d{8}/)"
+          },
+          "title":{
+            ".validate": "newData.isString() && newData.val().length > 0 && newData.val().length < 256"
+          },
+          "uniqueId" : {
+            ".validate" : "newData.isString() && newData.val().length > 0"
+          },
+          "checkedDays" : {
+            "$dateString" : {
+              ".validate" : "$dateString.matches(/\\d{8}/) && (newData.val() === 1 || newData.val() === 0)"
+            }
+          }
+       },
         "writers" : {
-          ".read" : "auth !== null && data.child(auth.uid).exists()",
-          ".write" : "auth !== null && data.child(auth.uid).exists()"
+          ".write" : "auth !== null && (!data.exists() || root.child('calendars/' + $calendarId + '/writers/' + auth.uid).exists())",
```

Comment:
You're already here.  `data.child(auth.uid).exists()` ?  Maybe `data.parent.child` at worst.